File: DotNet/SETPractice/DesignPattern/BuilderPattern/CarBuilder.cs
namespace DesignPattern.BuilderPattern
{
    public class CarBuilder : ICarBuilder
    {
        public Car car = new Car();

        public void SetBody()
        {
            Console.WriteLine("Car body assembled");
        }

        public void SetDieselEngine()
        {
            Console.WriteLine("Diesel Engine assembled");
        }

        public void SetHeadLights(int headLights)
        {
            car.HeadLights = headLights;
            Console.WriteLine($"{car.HeadLights} Headlights assembled");
        }

        public void SetPetrolEngine()
        {
            Console.WriteLine("Petrol Engine assembled");
        }

        public void SetSeats(int seats)
        {
            car.Seats = seats;
            Console.WriteLine($"{car.Seats} Seats assembled");
        }

        public Car GetCar() 
        {
            return car;
        }
    }
}

File: Java/SETPractice/src/main/java/DesignPattern/BuilderPattern/CarBuilder.java
package DesignPattern.BuilderPattern;

public class CarBuilder implements ICarBuilder{
    public Car car = new Car();

    @Override
    public void SetPetrolEngine() {
        System.out.println("Petrol Engine assembled");
    }

    @Override
    public void SetDieselEngine() {
        System.out.println("Diesel Engine assembled");
    }

    @Override
    public void SetBody() {
        System.out.println("Car body assembled");
    }

    @Override
    public void SetSeats(int seats) {
        car.Seats = seats;
        System.out.println("Car seats assembled");
    }

    @Override
    public void SetHeadLights(int headLights) {
        car.HeadLights = headLights;
        System.out.println("Car Head lights assembled");
    }

    @Override
    public Car GetCar() {
        return car;
    }
}
